src/lib/types.ts

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+export interface JestFileResults {
+    name: string;
+    summary: string;
+    message: string;
+    status: "failed" | "passed";
+    startTime:number;
+    endTime:number;
+    assertionResults: JestAssertionResults[];
+}
+
+export interface JestAssertionResults {
+    name: string;
+    title: string;
+    status: "failed" | "passed";
+    failureMessages: string[];
+}
+
+export interface JestTotalResults {
```

Comment:
The nice thing is that we have all those types in Jest already: https://github.com/facebook/jest/tree/master/types so once we merge there'll be some good de-duplication of this and it'll be easier to keep them in sync going forward!